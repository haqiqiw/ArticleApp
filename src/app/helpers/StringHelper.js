export default class StringHelper {
  static parseISOString(s) {
    if (s.length == 0) return null;
    var b = s.split(/\D+/);
    return new Date(Date.UTC(b[0], --b[1], b[2], b[3], b[4], b[5], b[6])).toDateString();
  }
}