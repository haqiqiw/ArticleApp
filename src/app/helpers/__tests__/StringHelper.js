import StringHelper from '../StringHelper';

describe('Parse ISO String', () => {
  it('should return the string of iso date (2018-05-16T15:30:15+0000 = Wed May 16 2018)', () => {
    expect(StringHelper.parseISOString('2018-05-16T15:30:15+0000')).toMatchSnapshot();
  });

  it('should return null', () => {
    expect(StringHelper.parseISOString('')).toMatchSnapshot();
  });
});