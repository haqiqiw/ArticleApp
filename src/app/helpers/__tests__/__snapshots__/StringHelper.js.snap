// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parse ISO String should return null 1`] = `null`;

exports[`Parse ISO String should return the string of iso date (2018-05-16T15:30:15+0000 = Wed May 16 2018) 1`] = `"Wed May 16 2018"`;
